### DISCUSSION TITLE
如何使用原生的require

### DISCUSSION THREAD

**Discussion by uummxx** [lisonge/vite-plugin-monkey#136] [Q&A] [answered]
我想要监听网路请求 这里使用到别人的库[ajaxHooker]( https://update.greasyfork.org/scripts/483208/1302155/ajaxHooker_myaijarvis.js)，我需要如何调用他呢？
我都不能打印 这个函数
![Clip_2024-02-29_20-29-24](https://github.com/lisonge/vite-plugin-monkey/assets/97946163/0b2a910c-a105-48e3-abd6-0ad6df4af34b)

![Clip_2024-02-29_20-28-03](https://github.com/lisonge/vite-plugin-monkey/assets/97946163/e3e6d1ab-1119-41f9-80b6-28e2769e8809)
然后我测试打包，他也会提示
![Clip_2024-02-29_20-30-39](https://github.com/lisonge/vite-plugin-monkey/assets/97946163/19dfa51c-994b-43a8-8558-1c153760d913)

**Comment by lisonge** [comment]
在这个链接的后面添加一个将 `;window.ajaxHooker=ajaxHooker;` 编码为 dataUri 的字符串
然后在代码里通过 `import {monkeyWindow} from "$"; monkeyWindow.ajaxHooker`  使用

**Reply by uummxx** [reply]
我已经这样子用了，但是它提示undefined
![Clip_2024-02-29_21-29-35](https://github.com/lisonge/vite-plugin-monkey/assets/97946163/4f7768e7-481b-4615-ada1-9cc3457d0b62)

**Reply by lisonge** [reply]
>  编码为 dataUri 的字符串

**Reply by lisonge** [reply, marked as answer]
```ts
import { defineConfig } from 'vite';
import monkey, { util } from 'vite-plugin-monkey';

export default defineConfig({
  plugins: [
    monkey({
      entry: 'src/main.ts',
      userscript: {
        require: ['your_url', util.dataUrl(';window.ajaxHooker=ajaxHooker;')],
      },
    }),
  ],
});

```
